blocks/contact-form/contact-form.js

```diff
@@ -69,12 +70,24 @@ export default function decorate(block) {
     }
   });
   // Specify submitContactInfo arguments
-  const submitInfo = (event) => {
+  const submitInfo = async (event) => {
     // To prevent the button from scrolling up when clicked
     if (event) {
       event.preventDefault();
     }
-    submitContactInfo(formInputs);
+
+    // Make sure no inputs are empty
+    const emptyInputs = [...formInputs].filter(
+      (input) => input.required && !input.value.trim(),
+    );
+
+    if (emptyInputs.length > 0) {
+      emptyInputs.forEach((input) => input.classList.add('required-empty-input'));
+      return;
+    }
```

Comment:
not a bad way to go about it. a more extendable way (to handle further validation rules besides required) would be to use `form.checkValidity()`

> checkValidity(): Returns true if the element's value has no validity problems; false otherwise. If the element is invalid, this method also fires an [invalid event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/invalid_event) on the element.
https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation#validating_forms_using_javascript